### DISCUSSION TITLE
why we set excutor =nullptr  in  WholeStageResultIterator::createNewVeloxQueryCtx()

### DISCUSSION THREAD

**Discussion by HaoChen-ch** [apache/gluten#6164] [Q&A] [answered]
https://github.com/apache/incubator-gluten/blob/79e1d588dcad5a655c2e5d363d18f8d695ef1cee/cpp/velox/compute/WholeStageResultIterator.cc#L177
why we set excutor =nullptr  in  WholeStageResultIterator::createNewVeloxQueryCtx()? is that mean all tasks is run on a single thread?

**Comment by zhztheplayer** [comment, marked as answer]
That's correct. We basically follow Spark's threading model that one thread per task.

**Reply by HaoChen-ch** [reply]
thanks very much.I got it
